### PR TITLE
Prepare v1.0.0

### PR DESCRIPTION
**Pull request by nepalez**
This is a new reincarnation of the gem. I totally re-wrote its the very
architecture for consistency and unification reasons. This version is
backward-incompatible to all the previous ones up to the [0.3.3].

Its DSL was re-written and regularized as well. While the idea of the gem
remains about the same as before, its implementation has been changed
in many important details.

See the CHANGELOG for details

### CODE REVIEW

**Comment by palkan** [comment]
Oops, I'm so sorry; I've opened the wrong tab to close a PR(

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,254 @@ All notable changes to this project will be documented in this file.
...
+  MyClient.new
+          .scopes[:users][version: "1.1"]
+          .operations[:fetch][id: 7]
+          .call
+  ```
+
+  You can use more natural one:
+
+  ```ruby
+  MyClient.new.users(version: "1.1").fetch(id: 7)
```
Comment (palkan) [inline CHANGELOG.md]: ❤️

File: CHANGELOG.md [hunk 2]
```diff
@@ -36,7 +279,7 @@ On the road to v0.4.0
 
 This version changes the way of processing responses. Instead of dealing
 with raw rake responses, we add opinionated methods to gracefully process
```
Comment (palkan) [inline CHANGELOG.md]: Typo? `rack responses`

File: docs/rspec.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+When you provide a client to remote API, you would provide some means for its users to test their operations.
+
+Surely, they could use [webmock] to check the ultimate requests that are sent to the server. But doing this, they would inadvertedly specify not their own code, but your client's code too. What do they actually need is a means to check calls of your client's operations. This way they would back on correctness of your client, and take its interface as an endpoint.
+
+For this reason, we support a special RSpec matcher `perform_operation`. It checks, what operations are called via evil-client, and what options are used in there.
```
Comment (palkan) [inline docs/rspec.md]: What about adding some support for integration tests too? I mean something like that:

```ruby
before { stub_operation(CatClient, "cats.fetch") }

# or

before { stub_operation(CatClient, /cats\..*/) }

# or with return value
before { stub_operation(CatClient, /.*/).and_return(double("Cat")) }

# or with constraints
before { stub_operation(CatClient, "cats.fetch").with(token: "valid") }

# etc
```

Now I'm testing evil-client integration the following way:

```ruby
let(:candidates) do 
  double('candidates').tap do |stub|
    allow(client).to receive(:candidates).and_return(stub)
  end
end

it "works" do
  expect(candidates).to receive(:create).with(
        ...
   ).and_return(double('response', id: 'checkr-123'))

  # check some code that handles the response
end
```

Comment (nepalez) [inline docs/rspec.md]: I'm going to return to this place in a week. For now, I just copy-pasted your comment into the separate [issue](https://github.com/evilmartians/evil-client/issues/18)

**Pull request merged**